src/OpenLoco/src/Map/MapGenerator/MapGenerator.cpp

```diff
@@ -278,40 +278,39 @@ namespace OpenLoco::World::MapGenerator
             if (surface == nullptr)
                 continue;
 
-            if (surface->isIndustrial())
+            if (!surface->isIndustrial())
             {
                 auto* landObj = ObjectManager::get<LandObject>(surface->terrain());
                 if (landObj->hasFlags(LandObjectFlags::unk0))
                 {
+                    bool setVariation = false;
                     if (surface->water())
                     {
-                        auto waterHeight = surface->waterHeight();
+                        auto waterLevel = surface->water();
                         if (surface->slope())
-                            waterHeight -= kSmallZStep;
+                            waterLevel -= 4;
 
-                        if (waterHeight > surface->baseZ())
+                        if (waterLevel > surface->baseZ())
```

Comment:
This still looks wrong i think. Haven't checked but pretty sure baseZ needs a kSmallMicro divide